packages/infinite-scroll/src/main.js

```diff
@@ -84,13 +84,20 @@ const getScrollOptions = (el, vm) => {
   }, {});
 };
 
-const getElementTop = el => el.getBoundingClientRect().top;
+const getElementTop = el => getElementContainer(el).getBoundingClientRect().top;
+
+const getElementContainer = el => el === window ? window.document.body : el;
+
+const isElementVisible = el => {
+  let rect = getElementContainer(el).getBoundingClientRect();
+  return rect.width > 0 && rect.height > 0;
+};
 
 const handleScroll = function(cb) {
   const { el, vm, container, observer } = this[scope];
   const { distance, disabled } = getScrollOptions(el, vm);
 
-  if (disabled) return;
+  if (disabled || !isElementVisible(el)) return;
 
```

Comment:
i suggest to create `IntersectionObserver` to watch element visible status when invisible, like this:
```
if (!isElementVisible(el) && !visibleObserver) {
	visibleObserver = new IntersectionObserver(onScroll);
	visibleObserver.observe(el);
}
```